db/1.9/modules/ROOT/edits/database-connector-query-single.adoc

```diff
@@ -0,0 +1,215 @@
+. Set the *Path* field to `/select/book/{maxId}`. +
+The `maxId` value parameter indicates how many books records to retrieve from the database. You can increment this value regardless of the number of maximum connections you configured in your database connection pool.
+. In the *Mime Type* tab, set the *Mime Type* field to `application/json`.
+. In the *Advanced* tab, set the *Allowed methods* field to `GET`.
+. Drag a *Set Variable* component to the right of the *Listener* source. +
+This component creates a new variable to save the database results that will be obtained from the *Query single* operation.
+. Set the *Name* field to `bookCollection` and the *Value* field to `#[[]]`.
+
+.Set Variable configuration
+image::database-querysingle-example-3.png[Set Variable configuration]
+
+=== Configure the For Each Component, the Query Single Operation, and Set Payload Component
+
+Continue creating the Mule application using a *For Each* component to iterate over the number of books requested by *HTTP Listener*. Then retrieve the book information using the *Query single* operation, and save the results in a payload variable:
+
+. Drag a *For Each* component to the right of *Set Variable*. +
+. Set the *Collection* field to `#[1 to attributes.uriParams.maxId as Number]`. +
+This expression iterates the collection of books in the table, from the first book (`1`) to the number of books set in the `maxId` parameter, when performing the HTTP request `/select/book/{maxId}`.
++
+.For Each configuration
+image::database-querysingle-example-4.png[For Each configuration]
++
+[start=3]
+. Drag the *Query single* operation inside the *For Each* component.
+. Set the *Connector configuration* field to `Database_Config` to connect to the MySQL database configuration.
+. Set the *SQL Query Text* field to `SELECT id, title, author FROM BOOK WHERE id = :id`. +
+This query selects the books information from the database.
+. Set the *Input Parameters* field to `![CDATA[#[{'id': payload }]]]`. +
+This expression maps the key parameter `id` (referenced in the previous *SQL Query Text* expression) to `payload` which is the result value of books information retrieved using the `SELECT` query.
++
+.Query single configuration
+image::database-querysingle-example-5.png[Query single configuration]
++
+[start=7]
+. Drag another *Set Variable* component to the right of the *Query single* operation.
+. Set the *Name* field to `bookCollection` and the *Value* field to `#[vars.bookCollection ++ [payload]]`. +
+The original variable `bookCollection` now saves the `payload` of the retrieved book information.
+. Drag a *Set Payload* component to the right of the *For Each* component.
+. Set the *Value* field to `#[vars.bookCollection]` to save the variable content as a new output payload.
+
+.Set Payload configuration
+image::database-querysingle-example-6.png[Set Payload configuration]
+
+=== Run and Test the Mule Application
```

Comment:
run and test
vs.
complete and test in the first sentence
vs.
save and test in the task steps